activerecord/lib/active_record/relation/delegation.rb

```diff
@@ -109,12 +109,10 @@ def relation_class_for(klass)
         end
     end
 
-    def respond_to_missing?(method, include_private = false)
-      super || @klass.respond_to?(method, include_private) ||
-        arel.respond_to?(method, include_private)
-    end
-
     private
+      def respond_to_missing?(method, _)
+        super || @klass.respond_to?(method) || arel.respond_to?(method)
```

Comment:
I think that should not be passed because using `public_send` in `method_missing` and can't call `relation.send(:klass_private_method)` even if `relation.respond_to?(:klass_private_method, true)` is true.